kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/InCommitTimestampSuite.scala

```diff
@@ -0,0 +1,269 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults
+
+import io.delta.kernel.Operation.{CREATE_TABLE, WRITE}
+import io.delta.kernel._
+import io.delta.kernel.data.FilteredColumnarBatch
+import io.delta.kernel.engine.Engine
+import io.delta.kernel.expressions.Literal
+import io.delta.kernel.internal.actions.SingleAction
+import io.delta.kernel.internal.fs.Path
+import io.delta.kernel.internal.util.{Clock, FileNames, ManualClock, SystemClock}
+import io.delta.kernel.internal.{TableConfig, TableImpl}
+import io.delta.kernel.internal.util.Utils.singletonCloseableIterator
+import io.delta.kernel.types.IntegerType.INTEGER
+import io.delta.kernel.types._
+import io.delta.kernel.utils.CloseableIterable.{emptyIterable, inMemoryIterable}
+
+import java.util.Optional
+import scala.collection.JavaConverters._
+import scala.collection.immutable.Seq
+import scala.collection.mutable
+
+import io.delta.kernel.internal.TableConfig.IN_COMMIT_TIMESTAMPS_ENABLED;
+
+class InCommitTimestampSuite extends DeltaTableWriteSuiteBase {
+  private def getInCommitTimestamp(engine: Engine, table: Table, version: Long): Option[Long] = {
+    val logPath = new Path(table.getPath(engine), "_delta_log")
+    val file = engine
+      .getFileSystemClient
+      .listFrom(FileNames.listingPrefix(logPath, version)).next
+    val row =
+      engine.getJsonHandler.readJsonFiles(
+        singletonCloseableIterator(file),
+        SingleAction.FULL_SCHEMA,
+        Optional.empty()).toSeq.map(batch => batch.getRows.next).head
+    val commitInfoOrd = row.getSchema.indexOf("commitInfo")
+    val commitInfoRow = row.getStruct(commitInfoOrd)
+    val inCommitTimestampOrd = commitInfoRow.getSchema.indexOf("inCommitTimestamp")
+    if (commitInfoRow.isNullAt(inCommitTimestampOrd)) {
+      None
+    } else {
+      Some(commitInfoRow.getLong(inCommitTimestampOrd))
+    }
+  }
+
+  test("Enable ICT on commit 0") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val beforeCommitAttemptStartTime = System.currentTimeMillis
+      val table = Table.forPath(engine, tablePath)
+      val txnBuilder = table.createTransactionBuilder(engine, testEngineInfo, CREATE_TABLE)
+
+      val txn = txnBuilder
+        .withSchema(engine, testSchema)
+        .withTableProperties(engine, Map(IN_COMMIT_TIMESTAMPS_ENABLED.getKey -> "true").asJava)
+        .build(engine)
+
+      txn.commit(engine, emptyIterable())
+      val ver0Snapshot = table.getSnapshotAsOfVersion(engine, 0)
+      assert(TableConfig.IN_COMMIT_TIMESTAMPS_ENABLED.fromMetadata(ver0Snapshot.getMetadata))
+      assert(getInCommitTimestamp(engine, table, 0).get >= beforeCommitAttemptStartTime)
+      assert(getInCommitTimestamp(engine, table, 0).get == ver0Snapshot.getTimestamp(engine))
```

Comment:
Can these be verified by reading the commit file directly?

Reply:
What do you mean by reading commit file directly?